kernel/src/actions/visitors.rs

```diff
@@ -268,6 +251,59 @@ impl DataVisitor for RemoveVisitor {
     }
 }
 
+/// Get a DV out of some engine data. The column holding `storage_type` is specified as `storage_type_col_index`. We assume the calling code has
+/// extracted `storage_type` already to verify the DV data is there.
+pub(crate) fn visit_deletion_vector_at<'a>(
+    row_index: usize,
+    storage_type_col_index: usize,
+    storage_type: String,
+    getters: &[&'a dyn GetData<'a>],
+) -> DeltaResult<DeletionVectorDescriptor> {
+    let mut idx = storage_type_col_index + 1;
+    let path_or_inline_dv: String = getters[idx].get(row_index, "deletionVector.pathOrInlineDv")?;
+    idx += 1;
+    let offset: Option<i32> = getters[idx].get_opt(row_index, "deletionVector.offset")?;
+    idx += 1;
+    let size_in_bytes: i32 = getters[idx].get(row_index, "deletionVector.sizeInBytes")?;
+    idx += 1;
+    let cardinality: i64 = getters[idx].get(row_index, "deletionVector.cardinality")?;
+    Ok(DeletionVectorDescriptor {
+        storage_type,
+        path_or_inline_dv,
+        offset,
+        size_in_bytes,
+        cardinality,
+    })
+}
+
+#[derive(Default)]
+pub(crate) struct DeletionVectorVisitor {
+    pub(crate) descriptor: Option<DeletionVectorDescriptor>,
+}
+
+impl DeletionVectorVisitor {
+    pub(crate) fn visit_deletion_vector<'a>(
+        row_index: usize,
+        storage_type: String,
+        getters: &[&'a dyn GetData<'a>],
+    ) -> DeltaResult<DeletionVectorDescriptor> {
+        visit_deletion_vector_at(row_index, 0, storage_type, getters)
+    }
+}
+
+impl DataVisitor for DeletionVectorVisitor {
+    fn visit<'a>(&mut self, row_count: usize, getters: &[&'a dyn GetData<'a>]) -> DeltaResult<()> {
+        for i in 0..row_count {
+            // Since path column is required, use it to detect presence of an Add action
+            if let Some(storage_type) = getters[0].get_opt(i, "deletionVector.storage_type")? {
+                self.descriptor = Some(Self::visit_deletion_vector(i, storage_type, getters)?);
+                break;
```

Comment:
Who uses this, that would loop over the entire data chunk but `break` out at first match?
```suggestion
            if let Some(descriptor) = Self::visit_deletion_vector(i, storage_type, getters)? {
                self.descriptor = Some(descriptor);
                break;
```

Reply:
did something similar